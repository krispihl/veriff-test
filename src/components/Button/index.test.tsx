import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Button } from './index';

describe('Button', () => {
    it('renders Button component correctly', () => {
        const { container } = render(<Button disabled={false} onClick={jest.fn()}>Test</Button>);

        expect(container).toMatchSnapshot();
        expect(screen.getByText('Test', { exact: true})).toBeInTheDocument();
    });

    it('renders disabled Button component correctly', () => {
        const { container } = render(<Button disabled={true} onClick={jest.fn()}>Test</Button>);

        expect(container).toMatchSnapshot();
        expect(screen.getByText('Test', { exact: true})).toBeDisabled();
    });
})