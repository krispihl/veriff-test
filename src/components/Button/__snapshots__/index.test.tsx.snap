// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button component correctly 1`] = `
<div>
  <button
    class="button"
  >
    Test
  </button>
</div>
`;

exports[`Button renders disabled Button component correctly 1`] = `
<div>
  <button
    class="button"
    disabled=""
  >
    Test
  </button>
</div>
`;
